softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/handler/SmsConfirmRequestHandler.java

```diff
@@ -17,16 +29,95 @@
         extends AbstractRequestHandler<SmsConfirmRequest,SmsConfirmResponse>
         implements AuthorizationRequestHandler {
 
+    @Autowired
+    private AuthorizationCacheService cacheService;
+
+    @Autowired
+    private TokenService tokenService;
+
+    @Autowired
+    private ProfileService profileService;
+
+    @Autowired
+    private DeviceService deviceService;
+
     @Override
     public String getName() {
         return "sms_confirm";
     }
 
     @Override
     public Response<SmsConfirmResponse> doHandle(Request<SmsConfirmRequest> msg) {
-        // todo implement later
+        Response<SmsConfirmResponse> response = new Response<>();
+
+        String authCode = msg.getData().getAuthCode();
+        String regRequestUuid = msg.getData().getRegistrationRequestUuid();
+
+        RegistrationCacheData regData = cacheService.get(regRequestUuid);
+
+        if (regData != null && (regData.getAuthCode().equals(authCode))){
+
+            ProfileEntity profile = createProfile(regData);
```

Comment:
@Dominys later I implement method checkIfExist(String phone) nd fix this problem